pennylane/workflow/construct_batch.py

```diff
@@ -64,22 +108,49 @@ def _validate_level(
         ValueError: If the level is not recognized
     """
 
-    if isinstance(level, (int, slice)):
-        return
-
-    if isinstance(level, str):
-        if level not in ("top", "user", "device", "gradient"):
-            raise ValueError(
-                f"level {level} not recognized. Acceptable strings are "
-                "'device', 'top', 'user', and 'gradient'."
-            )
+    if isinstance(level, (int, slice, str)):
         return
 
     raise ValueError(f"level {level} not recognized. Acceptable types are int, str, and slice.")
 
 
+def _validate_custom_levels(program):
+    protected_options = {"top", "user", "gradient", "device", "all", "all-mlir"}
```

Comment:
Should we escalate this to a top level frozen set e.g. `PROTECTED_LEVELS = frozenset(...)`